src/linux/frida-helper-glue.c

```diff
@@ -1259,11 +1444,12 @@ frida_remote_write (pid_t pid, GumAddress remote_address, gconstpointer data, gs
     src++;
   }
 
-  dst = remote_address + (size / sizeof (gsize));
+  dst = remote_address + ((size / sizeof (gsize)) * sizeof (gsize));
+  src = data + ((size / sizeof (gsize)) * sizeof (gsize));
```

Comment:
Nice catch. How about extracting the repeated part here into a `remainder_offset` variable or something like that?